Sources/StreamChat/ChatClient.swift

```diff
@@ -404,6 +405,48 @@ public class ChatClient {
             completion()
         }
     }
+    
+    // MARK: - Listening for Client Events
+    
+    /// Subscribes to web-socket events of the specified event type.
+    ///
+    /// - Note: The handler is always called on the main thread.
+    ///
+    /// An example of observing connection status changes:
+    /// ```swift
+    /// client.subscribe(to: ConnectionStatusUpdated.self) { connectionEvent in
+    ///     switch connectionEvent.connectionStatus {
+    ///         case .connected:
+    ///           …
+    ///     }
+    /// }
+    /// ```
+    ///
+    /// - SeeAlso: ``Chat.subscribe(to:handler:)`` for subscribing to channel specific events.
+    ///
+    /// - Parameters:
+    ///   - event: The event type to subscribe to (e.g. ``ConnectionStatusUpdated``).
+    ///   - handler: The handler closure which is called when the event happens.
+    ///
+    /// - Returns: A cancellable instance, which you use when you end the subscription. Deallocation of the result will tear down the subscription stream.
+    @available(iOS 13.0, *)
+    public func subscribe<T>(to eventType: T.Type, handler: @escaping (T) -> Void) -> AnyCancellable where T: Event {
+        eventNotificationCenter.subscribe(to: T.self, handler: handler)
+    }
```

Comment:
`ChatConnectionController` is currently the way of observing connection events. But since observing connection events is same as observing any events, there is no point of having separate subscribe methods. Instead, I just added a snippet for that to the documentation (since people who migrate might get confused about it).

Reply:
Yeah the connection controller has always been very confusing, so Ideally we should not have it in the new state layer 👍